app/controllers/brexit_feedback_controller.rb

```diff
@@ -0,0 +1,58 @@
+require 'date'
+
+require 'google/apis/drive_v2'
+require 'google/api_client/client_secrets'
+
+require 'signet/oauth_2/client'
+
+class BrexitFeedbackController < ApplicationController
+  def home
+    session.destroy
+  end
+
+  def results
+    client_hash = session['auth_client']
+
+    opts = {
+            authorization_uri: "#{client_hash['authorization_uri']['scheme']}://#{client_hash['authorization_uri']['host']}#{client_hash['authorization_uri']['path']}",
+            token_credential_uri: "#{client_hash['token_credential_uri']['scheme']}://#{client_hash['token_credential_uri']['host']}#{client_hash['token_credential_uri']['path']}",
+            client_id: client_hash['client_id'],
+            client_secret: client_hash['client_secret'],
+            scope: client_hash["scope"][0],
+            redirect_uri: "#{client_hash['redirect_uri']['scheme']}://#{client_hash['redirect_uri']['host']}#{client_hash['redirect_uri']['path']}"
+           }
+
+    auth_client = Signet::OAuth2::Client.new(opts)
+
+    auth_client.code = params['code']
+    auth_client.additional_parameters = { access_type: 'offline', include_granted_scopes: 'true' }
+    brexit_slugs = Support::Requests::BrexitSlugFetcher.new(auth_client).slugs
+    @results = Support::Requests::BrexitFeedbackRequest.new(Date.parse(session['from_date']), Date.parse(session['to_date']), brexit_slugs).formatted_results
+  end
+
+  def auth
+    secrets = JSON.parse(File.read('client_secrets.json'))
```

Comment:
It might be simpler to make this file an JSON ERB template.
Then you can interpolate environment variables as you read the file and there's no need to rewrite the file:
`config/client_secrets.json.erb`:

```
{
  "web": {
    "client_id": <%= ENV.fetch("google_client_id", "client_id") %>,
    "client_secret": <%= ENV.fetch("google_client_secret", "client_secret") %>,
...
}
```

And then you can read/interpolate vars like so:

```ruby
template = File.read('config/client_secrets.json')
secrets = JSON.parse(ERB.new(template).results)
```

Reply:
nice, didn't know this was a thing. Thanks :)

Reply:
So the google auth client has to read from a file rather than being passed one. I'm going to take the approach that is currently implemented and move it around a bit to block any accidental uploads of secrets and use the erb templating but it annoyingly has to be saved.

Reply:
https://www.rubydoc.info/github/google/google-api-ruby-client/Google%2FAPIClient%2FClientSecrets.load takes a filename as an argument, this means you can do the ERB template dance and write the file out (as a Tempfile perhaps as it will contain real secrets) and pass the path of the tempfile to `Google::APIClient::ClientSecrets.load(filename)`